python/tests/simplify.py

```diff
@@ -158,13 +159,7 @@ def record_node(self, input_id, is_sample=False):
         flags &= ~tskit.NODE_IS_SAMPLE
         if is_sample:
             flags |= tskit.NODE_IS_SAMPLE
-        output_id = self.tables.nodes.add_row(
-            flags=flags,
-            time=node.time,
-            population=node.population,
-            metadata=node.metadata,
-            individual=node.individual,
-        )
+        output_id = self.tables.nodes.append(replace(node, flags=flags))
```

Comment:
I have to say I'm not a big fan of this pattern. It seems awkward to me to have to know that the rows are dataclasses, and then you need to call this function ``dataclasses.replace`` with the stuff you want to change. It's assuming that users are well versed in dataclasses.

What if we have a method
```python
def replace(self, **kwargs):
     return dataclasses.replace(self, **kwargs)
```
on the row classes so that this line would look like
```python
output_id = self.tables.nodes.append(node.replace(flags=flags))
```

at least then the recommended pattern is discoverable and we can document it locally.

Reply:
Yeah, I see what you mean. I'm not sure why those methods aren't on dataclasses in the first place. I've added a method by inheritance of a class in `util`